openpdf/src/main/java/com/lowagie/text/xml/simpleparser/SimpleXMLParser.java

```diff
@@ -550,7 +548,39 @@ public static void parse(SimpleXMLDocHandler doc, SimpleXMLDocHandlerComment com
         SimpleXMLParser parser = new SimpleXMLParser(doc, comment, html);
         parser.go(r);
     }
-    
+
+    /** Detect charset from BOM, as per <a href="https://unicode.org/faq/utf_bom.html#bom4">Unicode FAQ</a>. */
+    private static String detectCharsetFromBOM(byte[] bom) {
```

Comment:
Consider returning `Optional<Encoding>` so 1) return value will become more typed and 2) returning of null from enclosing method can be postponed

Reply:
Makes sense… so far this was a "minimal impact" style PR, but it makes sense to refactor the code a bit, while at it. (also, the two remaining `if`s make little sense, now that the EBCDID dead code was dropped)